modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java

```diff
@@ -3236,6 +3236,57 @@ else if (file.exists()) {
         return size;
     }
 
+    /**
+     * Dumps WAL segments since last known checkpoint.
+     */
+    public void dumpWalFiles() {
+        try {
+            long lastCpIdx = lastCheckpointPtr.index();
+
+            long currIdx = currentHandle().getSegmentId();
+
+            String path = "db/dump/" + igCfg.getConsistentId() + "/wal";
+            File dumpDir = U.resolveWorkDirectory(igCfg.getWorkDirectory(), path, false);
+
+            // bit set for tracking which file was copied to evade copying same segment twice from archive and work dir.
+            boolean[] copied = new boolean[(int) (currIdx - lastCpIdx + 1)];
+
+            List<File> walFiles = new ArrayList<>();
+
+            if (archiver != null)
+                walFiles.addAll(Arrays.asList(walArchiveDir.listFiles(WAL_SEGMENT_COMPACTED_OR_RAW_FILE_FILTER)));
+
+            walFiles.addAll(Arrays.asList(walWorkDir.listFiles(WAL_SEGMENT_COMPACTED_OR_RAW_FILE_FILTER)));
+
+            if (archiver != null) {
```

Comment:
What if archiver is null?

Reply:
Then we don't need to check archived segments, don't we?